src/Common/RemoteProxyConfigurationResolver.h

```diff
@@ -10,6 +10,19 @@
 namespace DB
 {
 
+struct ConnectionTimeouts;
+
+struct RemoteProxyHostFetcher
```

Comment:
Created this interface so the HTTP response can be mocked and some level of unit testing can be achieved